.pipelines/cni/k8s-e2e/k8s-e2e-job-template.yaml

```diff
@@ -0,0 +1,166 @@
+            # explictly unzip and strip directories from ginkgo and e2e.test
+            tar -xvzf kubernetes-test-linux-amd64.tar.gz --strip-components=3 kubernetes/test/bin/ginkgo kubernetes/test/bin/e2e.test
+
+        displayName: "Setup Environment"
+      - ${{ if eq(parameters.datapath, true) }}:
+        - template: ../k8s-e2e/k8s-e2e-step-template.yaml
+          parameters:
+            testName: Datapath
+            name: datapath
+            clusterName: ${{ parameters.clusterName }}
+            ginkgoFocus: '(.*).Networking.Granular|(.*)kubernetes.api'
+            ginkgoSkip: 'SCTP|hostNetwork'
+            os: ${{ parameters.os }}
+            processes: 4
+            attempts: 10
+      - ${{ if eq(parameters.netpol, true) }}:
+        - template: ../k8s-e2e/k8s-e2e-step-template.yaml
+          parameters:
+            testName: Network Policy
+            name: netpol
+            clusterName: ${{ parameters.clusterName }}
+            ginkgoFocus: '(.*).Netpol'
+            ginkgoSkip: 'SCTP'
+            os: ${{ parameters.os }}
+            processes: 1
+            attempts: 0
+      - ${{ if eq(parameters.dns, true) }}:
+        - template: ../k8s-e2e/k8s-e2e-step-template.yaml
+          parameters:
+            testName: DNS
+            name: dns
+            clusterName: ${{ parameters.clusterName }}
+            ginkgoFocus: '\[sig-network\].DNS.should'
+            ginkgoSkip: 'resolv'
+            os: ${{ parameters.os }}
+            processes: 4
+            attempts: 3
+      - ${{ if eq(parameters.container, true) }}:
+        - template: ../k8s-e2e/k8s-e2e-step-template.yaml
+          parameters:
+            testName: Container Conformance
+            name: container
+            clusterName: ${{ parameters.clusterName }}
+            ginkgoFocus: '\[sig-node\].Pods'
+            ginkgoSkip: 'Slow'
+            os: ${{ parameters.os }}
+            processes: 4
+            attempts: 2
+      - ${{ if eq(parameters.discovery, true) }}:
+        - template: ../k8s-e2e/k8s-e2e-step-template.yaml
+          parameters:
+            testName: Resource Discovery
+            name: discovery
+            clusterName: ${{ parameters.clusterName }}
+            ginkgoFocus: '\[sig-api-machinery\].Discovery'
+            ginkgoSkip: ''
+            os: ${{ parameters.os }}
+            processes: 4
+            attempts: 3
+      - ${{ if eq(parameters.liveness, true) }}:
+        - template: ../k8s-e2e/k8s-e2e-step-template.yaml
+          parameters:
+            testName: Liveness Probes
+            name: liveness
+            clusterName: ${{ parameters.clusterName }}
+            ginkgoFocus: '\[sig-node\].Probing'
+            ginkgoSkip: ''
+            os: ${{ parameters.os }}
+            processes: 4
+            attempts: 10 #
```

Comment:
Why 10 for this one ?

Reply:
Ping and curl is excessively flakey. The 10 attempts only re-runs the tests that fail 10 times, not the entire suite. It even lets you know how many times it flaked in the output!